tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/RedeployTest.java

```diff
@@ -448,6 +450,63 @@ public void testRedeployQueueDefaults() throws Exception {
       }
    }
 
+   @Test
+   public void testUndeployDivert() throws Exception {
```

Comment:
@michaelpearce-gain  / @jsmucr any chance you can look at why this test is failing?

if you run it as part of the test suite (mvn -Ptests test) it breaks most of the time.

And this is blocking me from releasing.


Another point is the clash of the user redeployment as it's clashing with a previous definition.. see my PR.

Reply:
@clebertsuconic re the user in other pr, i think the issue is the test that failed was enforcing you cannot clear a user, that test probably should have been changed. @jsmucr will have to send a pr tomorrow to revert your change and fix the test so that its legal to clear a user. That said if you need to release it can be sorted in separate ticket.

Re this one on diverts if @jsmucr doesnt get back to you, revert the PR

Reply:
@clebertsuconic @michaelandrepearce Thank you for the review, I'll get on with that.

Reply:
\*sad face\* 
My PR is wrong. What I originally wanted to implement was to reset the values when reading `broker.xml`. What I've done instead is that it resets the queue configuration every time an incomplete configuration object is passed to the `updateQueue` method. Which in fact breaks the `testUpdateQueueWithNullUser` test.

Reply:
@jsmucr if you could check and fix RedeployTest::testUndeployDivert and testRedeployQueueDefaults

with my latest PR merged 
org.apache.activemq.artemis.tests.integration.client.UpdateQueueTest.testUpdateQueueWithNullUser is passing.


But as you and Michael said something else seems to be going on.


I won't release 2.14 for another week, so we have time to fix this.

Reply:
I have no idea why testUndeployDivert fails... when I run it on my laptop it always pass. when I run it as part of mvn -Ptests test, it fails most of the time. (9 in 10). so I have not been successful on debugging it.

if you have no idea I will probably create a branch with lots of traces.

(perhaps the collector from your change fails with ConcurrentModificationException and redeployment does not work?) that's the only theory I have but I tried to prove it and I couldn't.

Reply:
@clebertsuconic I'll keep this in mind and try to resolve it later.

Your PR resolves this by reverting part of the functionality that I've implemented. It does not fix the problem as a whole.

For now I think it would be for the best to revert #3172 so neither the build or the configuration mechanism is broken. I shall come with a different approach later.